host/queryworkflow_test.go

```diff
@@ -613,7 +615,7 @@ func (s *IntegrationSuite) TestQueryWorkflow_Consistent_PiggybackQuery() {
 		} else if previousStartedEventID > 0 {
 			for _, event := range history.Events[previousStartedEventID:] {
 				if *event.EventType == types.EventTypeWorkflowExecutionSignaled {
-					handledSignal = true
+					handledSignal.Store(true)
```

Comment:
nit: do we need to be sure to set it just once? then please consider sync.Once

Reply:
I don't think it needs to be set just once.